src/ion/liveranges.rs

```diff
@@ -797,20 +757,20 @@ impl<'a, F: Function> Env<'a, F> {
         // when needed, here and then again at the end of allocation
         // when resolving moves.
 
-        for vreg in &mut self.vregs {
+        for vreg in self.vregs.iter_mut() {
```

Comment:
I suppose we could implement `IntoIterator` to make this work again (and whatever the equivalent is for mut iters) but I don't mind too much if you want to punt on that either...

Reply:
I added `IntoIterator` implementations mutable and non-mutable references, and reverted the places where I had switched to explicit use of `iter`/`iter_mut` :+1: